src/ucp/core/ucp_ep.c

```diff
@@ -1043,7 +1043,7 @@ static void ucp_ep_config_init_attrs(ucp_worker_t *worker, ucp_rsc_index_t rsc_i
 
     iface_attr = ucp_worker_iface_get_attr(worker, rsc_index);
 
-    if (iface_attr->cap.flags & short_flag && !context->num_mem_type_mds) {
+    if (iface_attr->cap.flags & short_flag && context->config.ext.enable_memtype_cache) {
```

Comment:
@bureddy does this check mean that if we disable memtype_cache explicitly (from cmd line) then max_short = -1, and short am will be disabled? Why do we need this check? Isn't the check of cache being empty (in the send flow) not enough?

Reply:
No. In send flow it does am short if cache is not created or if it is empty.  if cache is  disabled from cmd line, then that condition will be always true and try to send inline. this will segfault for cuda memory.